api/login_api.py

```diff
@@ -31,7 +30,10 @@ def do_get(self, **kwargs):
     self.abort(405, valid_methods=['POST'])
 
   def do_post(self, **kwargs):
-    token = self.get_param('credential')
+    request = SignInRequest.from_dict(self.request.json)
+    token = request.credential
+    if not token:
+        raise werkzeug.exceptions.BadRequest(description="Missing required field 'credential'")
```

Comment:
It might be worth catching the error `from_dict` will raise if they json does not match what is needed to create the object here (this is done in your previous changes in cues_api)

Reply:
good point. thanks!